[findingfaey] iP

findingfaey opened the pull request:
### ✨ **Arin Task Manager**  
> “Your mind is for having ideas, not holding them.” – David Allen [[(source)](https://gettingthingsdone.com/)](https://gettingthingsdone.com/)

Arin is a **text-based** task manager designed to help you stay productive by organizing your tasks efficiently. It's:

- 📝 **Easy to Use**
- 🚀 **Fast and Lightweight**
- 🔥 **Highly Extendable**

---

### 🎯 **How to Get Started**  

1. **[[Download the latest release](https://github.com/findingfaey/releases)](https://github.com/findingfaey/releases)**
2. Open a terminal and run:
   ```sh
   java -jar arin.jar
   ```
3. Start adding tasks instantly!
4. Manage your tasks with simple commands.
5. Let Arin do the heavy lifting for you! 😃

---

### 🛠 **Features**
- ✅ **Task Management**
- 📅 **Manage Deadlines**
- 🔍 **Search Tasks with Find Feature**
- 📌 **Reminders (Coming Soon)**

---

### 🚀 **For Developers**
If you are a **Java** programmer, you can use this project to practice Java! Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

---

### ✅ **Checklist**
- [x] Implement Find feature  
- [x] Write JUnit tests for Find feature  
- [x] Apply coding standards  
- [ ] Implement reminders (Upcoming)  

---

### ⚡ **Key Commands**
Use inline code for quick references:
- **Add a task:** `todo Read a book`
- **Mark as done:** `mark 1`
- **Delete a task:** `delete 2`
- **List all tasks:** `list`
- **Find a task:** `find book`

---

### 🔗 **More Info**
For a full list of commands and features, check out the [[official documentation](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

findingfaey committed:
Level-0: Implement chatbot greeting and exit functionality

findingfaey committed:
Level-0: Add chatbot greeting and exit functionality

findingfaey committed:
Level-1: Add echo feature

findingfaey committed:
Level-2: Add support for adding and listing tasks

findingfaey committed:
Level-3: Add support for marking tasks as done

findingfaey committed:
Level-4: Add support for ToDos, Deadlines, and Events

findingfaey committed:
A-Inheritance: Implement task types using inheritance

findingfaey committed:
A-TextUiTesting: Fix test automation setup

findingfaey committed:
Level-5: Implement error handling with A-Exceptions

findingfaey committed:
Level-5: Implement error handling with A-Exceptions

findingfaey committed:
Level 5: Handle errors and invalid inputs

findingfaey committed:
Level-6: Implement delete task functionality

findingfaey committed:
A-Enums: Use enumerations for task types but incomplete cos the ordering gets abit fucked grrrr

findingfaey committed:
stash

findingfaey committed:
correct A-Enums but still not ordered correctly

elizazaa commented on src/main/java/Arin.java:
```diff
@@ -1,41 +1,39 @@
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
```
Should these be part of another commit?

elizazaa commented on src/main/java/Arin.java:
```diff
@@ -1,41 +1,39 @@
                 System.out.println("____________________________________________________________");
                 System.out.println(" Bye. Hope to see you again soon!");
                 System.out.println("____________________________________________________________");
                 break;
```
I like how you followed with the default outputs as the question! I think it makes it easier to debug this way

elizazaa reviewed (commented):
I think it the commit could be broken down into smaller commits but otherwise good job! I encourage you to maybe change the commands/outputs that suits Arin's persona better

findingfaey committed:
Refactor Arin chatbot: improve formatting and output handling

findingfaey committed:
Level 7: Implement task saving/loading from file

findingfaey committed:
Level 8: Implement date and time parsing and handling

findingfaey committed:
Merge branch 'branch-Level-8'
Merge branch 'branch-Level-8' into master: Implement date and time handling for deadlines and events

findingfaey committed:
Level A-MoreOOP: Refactor code to follow more OOP principles

findingfaey committed:
Merge branch 'branch-A-MoreOOP'

findingfaey committed:
A-Packages: Organize classes into packages

findingfaey committed:
Merge branch 'branch-A-Packages'

findingfaey committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2425S2/ip into add-gradle-support

findingfaey committed:
Merge branch 'add-gradle-support'

findingfaey committed:
Level A-Gradle: Set up Gradle build system

findingfaey committed:
A-JUnit: Add JUnit tests and update project structure

findingfaey committed:
A-JavaDoc: Add Javadoc comments to classes and methods

findingfaey committed:
A-CodingStandard: Enforce coding standards across all classes

findingfaey committed:
Level-9: Add JUnit tests for find feature

findingfaey committed:
Merge branch-Level-9 into master: Implement Find feature

meglwhy commented on src/main/java/arin/task/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+        this.by = LocalDate.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+    }
+
+    @Override
```
I like the usage of line wrapping here where appropriate.

meglwhy commented on src/main/java/arin/task/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+        this.tasks = tasks;
+    }
+
+    /**
```
I like how you added header comments to all methods here

meglwhy commented on src/main/java/arin/ui/Ui.java:
```diff
@@ -0,0 +1,114 @@
+package arin.ui;
+
```
I like how you organized the imports here.

meglwhy reviewed (commented):
LGTM!
Just some minor nits. Atb!

findingfaey committed:
Implement GUI for Level-10 and add support for varargs (A-Varargs)

findingfaey committed:
Add assertions to enforce task constraints

Task creation currently does not have internal checks for invalid inputs,
which may lead to unintended behavior. While there are exception checks
for user input validation, adding assertions will help detect logical
errors during debugging.

Let's add assertions to ensure:
* Task descriptions are not empty.
* Deadlines and events have valid date formats.
* Event start time is always before the end time.
* Task list modifications use valid indices.

Using assertions ensures that invalid states never propagate deeper
into the application and helps catch unexpected bugs early.

Tag: A-Assertions

findingfaey committed:
Replace magic strings with constants in Parser

The Parser class contains multiple instances of hardcoded strings such as
" /by " and " /from ", which makes the code harder to maintain and prone to
errors if these delimiters change in the future.

Let's define constants for:
* DEADLINE_DELIMITER -> " /by "
* EVENT_DELIMITER -> " /from | /to "

Using named constants improves readability, makes future modifications
easier, and reduces the chances of introducing subtle bugs.

Tag: A-CodeQuality
mmitted:

findingfaey committed:
extra changes from previous commit

findingfaey committed:
Resolve merge conflict in Parser.java

findingfaey committed:
Implement Java Streams API for task filtering and operations

This update implements Stream API operations in the TaskList class to provide:
- Improved search capabilities with case-insensitive searching
- New methods for filtering tasks by status, type, and time periods
- Support for upcoming deadline and event tasks with flexible time ranges
- Sorting capabilities for organizing tasks by description and deadlines
- Group operations for categorizing tasks and generating statistics
- Multi-keyword search with both ANY and ALL matching modes
- Added assertions for increased code robustness

Also added:
- New FindUpcomingCommand that leverages Stream operations
- Updated Parser to handle the upcoming command with day parameters
- Unit tests for Stream implementations in TaskListStreamsTest

This implementation enhances performance for filter operations and
simplifies the codebase by replacing imperative code with declarative
Stream operations, making future extensions easier to implement.

findingfaey committed:
Add help command functionality

findingfaey committed:
Implement C-Sort extension

findingfaey committed:
Updated README

findingfaey committed:
Add User Guide and command documentation

A comprehensive user guide has been added with:
- Detailed command documentation with examples
- Screenshots demonstrating all major features
- Step-by-step usage instructions for core functionality
- Task management examples
- Search and sort feature demonstrations

Also fixed:
- UnmarkTaskCommand index handling
- Proper screenshot organization
- Documentation formatting and structure

findingfaey committed:
Enhance documentation with advanced markdown features

- Add table of contents with navigation links
- Create organized command tables
- Include collapsible sections with emojis
- Add detailed formatting tips
- Improve visual hierarchy
- Include blockquotes for important notes
- Add code blocks for command examples

findingfaey committed:
Configure Git to ignore user data files

Configuration changes:
- Add data/arin.txt to .gitignore
  * Prevent user task data from being tracked
  * Avoid unintentional commits of personal data
  * Keep repository clean of user-specific content

Benefits:
- Improves repository maintainability
- Protects user privacy
- Reduces unnecessary file tracking
- Prevents merge conflicts from local task data

Technical:
- Ignore pattern matches: data/arin.txt
- Applies to local task storage file only
- Maintains application functionality while improving Git workflow

Part of repository maintenance and best practices implementation

findingfaey committed:
Update user guide with comprehensive feature demonstrations

Documentation improvements:
- Add event creation and task listing examples
  * Include screenshot of adding event task
  * Show comprehensive task list view
  * Demonstrate event date-time formatting

- Complete sorting functionality documentation
  * Add screenshots for all sorting criteria
  * Demonstrate type-based sorting (ToDo → Deadline → Event)
  * Show status-based sorting (incomplete vs complete)
  * Present examples of different sorting results

These additions provide:
- Better visual guidance for new users
- Complete feature coverage in documentation
- Clear examples of all available sorting options
- Improved understanding of task management workflow

Part of A-UserGuide enhancement

findingfaey committed:
Fix documentation rendering and image path issues

- Update image URLs to use consistent raw.githubusercontent.com format
- Fix table formatting to ensure proper rendering in GitHub Pages
- Standardize HTML tags to improve cross-platform compatibility
- Adjust layout of task examples for better readability
- Remove extraneous formatting characters causing display issues

findingfaey committed:
Improve documentation readability and formatting

- Center all images for better presentation
- Replace tables with cleaner list format
- Standardize command descriptions
- Fix display issues for GitHub Pages rendering

findingfaey committed:
Improve UI color scheme and readability

- Replace bright green with softer olive green (#708238)
- Enhance header contrast for better logo visibility
- Maintain consistent color palette across components
- Improve overall visual comfort for extended use

findingfaey committed:
Implement assertion checks for method preconditions

- Add assertion for index bounds in getTask method
- Add null check assertion for search keyword in findTasks
- Add null check assertion for taskType parameter in getTasksByType
- Add non-negative check for days parameter in getTasksDueWithinDays
- Ensure runtime safety by validating inputs early
- Prevent potential bugs by making assumptions explicit
- Improve code quality and robustness

findingfaey committed:
Improve typography and readability

- Update font family to modern, clean sans-serif
- Adjust font size for better readability
- Apply consistent typography across components
- Enhance visual hierarchy with proper font weights
- Improve overall aesthetic for a more professional look

findingfaey committed:
Merge pull request #2 from findingfaey/font-style-improvements

Font Style Improvements: Improve typography and readability
